openstack/dns/v2/zones/results.go

```diff
@@ -0,0 +1,63 @@
+package zones
+
+import (
+	"encoding/json"
+	"github.com/gophercloud/gophercloud"
+	"github.com/gophercloud/gophercloud/pagination"
+)
+
+type commonResult struct {
+	gophercloud.Result
+}
+
+// Extract interprets a GetResult, CreateResult or UpdateResult as a concrete Zone.
+// An error is returned if the original call or the extraction failed.
+func (r commonResult) Extract() (Zone, error) {
+	var s Zone
+	err := r.ExtractInto(&s)
+	return s, err
+}
+
+// GetResult is the deferred result of a Get call.
+type GetResult struct {
+	commonResult
+}
+
+// ZonePage is a single page of Zonje results.
+type ZonePage struct {
+	pagination.LinkedPageBase
+}
+
+// IsEmpty returns true if the page contains no results.
+func (p ZonePage) IsEmpty() (bool, error) {
+	services, err := ExtractZones(p)
+	return len(services) == 0, err
+}
+
+// ExtractZones extracts a slice of Services from a Collection acquired from List.
+func ExtractZones(r pagination.Page) ([]Zone, error) {
+	var s struct {
+		Zones []Zone `json:"zones"`
+	}
+	err := (r.(ZonePage)).ExtractInto(&s)
+	return s.Zones, err
+}
+
+type Zone struct {
+	ID            string                          `json:"id,omitempty"`
+	PoolID        string                          `json:"pool_id,omitempty"`
+	ProjectID     string                          `json:"project_id,omitempty"`
+	Name          string                          `json:"name,omitempty"`
+	Email         string                          `json:"email,omitempty"`
+	TTL           int                             `json:"ttl,omitempty"`
+	Serial        json.Number                     `json:"serial,omitempty"`
```

Comment:
Because `json.Number` and the different time formats (like `JSONRFC3339MilliNoZ`) are kind of awkward to use, we're trying to move toward only having built-in data types in response fields (see #75). To that end, perhaps we could implement `json.Unmarshaler` (see [here](https://github.com/gophercloud/gophercloud/blob/master/openstack/imageservice/v2/images/results.go#L82) for an example for both `json.Number` and `time.Time`) for `Zone`?